src/model.jl

```diff
@@ -8,90 +9,180 @@ mutable struct Model
-
+  
     co_albedo = 1.0 .- albedo
     
     ecc = eccentricity(1950)
     ob = obliquity(1950)
     per = perihelion(1950)
-    solar_forcing = calc_solar_forcing(co_albedo, ecc=ecc, ob=ob, per=per) #TODO: Add arguments [nx, ny, num_steps_year]
+     
+    
+    lambda,solar = calc_solar_forcing_params(nlatitude=ny, ntimesteps=num_steps_year, ecc=ecc, ob=ob, per=per)
+    solar_forcing = calc_solar_forcing(co_albedo, solar,nx,ny,num_steps_year)
   
     
-    return Model(diffusion_coeff, heat_capacity, albedo, solar_forcing, radiative_cooling_co2, radiative_cooling_feedback,co_albedo,compute_albedo)
-end
-
-function set_co2_concentration!(model, co2_concentration)
+    return Model(diffusion_coeff, heat_capacity, albedo, solar_forcing, radiative_cooling_co2, radiative_cooling_feedback,co_albedo,compute_albedo,geography,solar, compute_sea_ice_extent,year)
+  end
+  
+  function set_co2_concentration!(model, co2_concentration)
     model.radiative_cooling_co2 = calc_radiative_cooling_co2(co2_concentration)
-end
-
-
+  end
+  
+  
+  # Gives the possibility to interactively decide which parameters should be updated in a simulation
 
-Base.size(model::Model) = size(model.heat_capacity)
+  function update_model(model, mesh, time_step, update_geography=true, update_albedo=true, update_diffusion=true,update_heat_capacity=true, update_solar_forcing=true)
+  @unpack nx,ny = mesh
+  
+  if update_geography == true
+    set_geography!(model,mesh, time_step)
+  end
+  
+  if update_albedo == true
+    set_albedo!(model,mesh)
+    set_co_albedo!(model)
+  end
+  
+  if update_diffusion == true
+    set_diffusion_coeff!(model, mesh)
+  end
+  if update_heat_capacity == true
+    set_heat_capacity!(model)
+  end
+  if update_solar_forcing == true
+    set_solar_forcing!(model, mesh)
+  end
+  end
+  
+  
+  function set_geography!(model, mesh, time_step)
+  @unpack nx,ny = mesh
+
+  # monthly geography can be updated by reading the input maps or by computing the sea ice extent
+  # first time step is the last week of March as the simulation starts at the vernal equinox
+   if model.compute_sea_ice_extent == false
+      if time_step in (1,46,47,48)
+        model.geography = read_geography(joinpath(@__DIR__, "..", "input","world", "year2004", string("The_World_from_image", nx, "x", ny,"_1")),nx,ny) 
+    
+      elseif mod(time_step,4) == 2 && time_step in (2:45)
+       month::Int64 = 1
+        for nt in 1:time_step
+          if mod(nt,4) == 1
+           month +=1
+          end
+        end
+        model.geography = read_geography(joinpath(@__DIR__, "..", "input","world", "year2004", string("The_World_from_image", nx, "x", ny,"_",month)),nx,ny)
+      end
+   else  
+     annual_sea_ice_extent = read_sea_ice_extent_N(model.year)
+     model.geography = calc_geography_per_month_extent(model.geography,annual_sea_ice_extent, time_step, nx, ny)
```

Comment:
```suggestion
     calc_geography_per_month_extent!(model.geography,annual_sea_ice_extent, time_step, nx, ny)
```
This is the right way to call a function that modifies one of its arguments. See also the suggestion that I added in the function.